source_http.go

```diff
@@ -43,13 +50,27 @@ func (s *HttpImageSource) fetchImage(url *url.URL) ([]byte, error) {
 		return nil, fmt.Errorf("Error downloading image: (status=%d) (url=%s)", res.StatusCode, req.URL.String())
 	}
 
+	// Read the body
 	buf, err := ioutil.ReadAll(res.Body)
 	if err != nil {
 		return nil, fmt.Errorf("Unable to create image from response body: %s (url=%s)", req.URL.String(), err)
 	}
 	return buf, nil
 }
 
+func (s *HttpImageSource) setAuthorizationHeader(req *http.Request, ireq *http.Request) {
+	auth := s.Config.Authorization
```

Comment:
In general, I tend to bet for explicit behavior, since it's more predictable and generally easier to understand. In this scenario, I would prefer keeping the existent behavior for the same reasons and security concerns. 

If you need multiple Authorization tokens per origin, you should expose the header explicitly from the client side, since it's the most flexible and simple way to do that, or use an intermediate proxy that would instrument the token in incoming traffic. The approaches are multiple and would depend in the particular scenario.

Typically, as you mentioned, an `imaginary` server that forwards headers should not be accessible to non-authorized clients from a public network in any case.